RHDEVDOCS 6346 configure PAC pipeline run cancel-in-progress, plus mo… (#88596)
openshift/openshift-docs: modules/op-speccancel-pipeline-run-using-pipelines-as-code.adoc

// This module is included in the following assemblies:
// * pac/managing-pipeline-runs-pac.adoc

:_mod-docs-content-type: REFERENCE
[id="speccancel-pipeline-run-using-pipelines-as-code_{context}"]
= Annotations for specifying automatic cancellation-in-progress for a pipeline run

> By default, {pac} does not cancel pipeline runs automatically. Every pipeline run that {pac} creates and starts executes until it completes. However, events that trigger pipeline runs can come in quick succession. For example, if a pull request triggers a pipeline run and then the user pushes new commits into the pull request source branch, each push triggers a new copy of the pipeline run. If several pushes happen, several copies can run, resulting in a drain on cluster resources.

Reviewer: This is a correct use of the term _drain_, I think. However, in the OCP docs, it seems that _drain_ is almost always (if not always) in reference to the MCO draining pods from a node. Maybe use a different word here?

```suggestion
By default, {pac} does not cancel pipeline runs automatically. Every pipeline run that {pac} creates and starts executes until it completes. However, events that trigger pipeline runs can come in quick succession. For example, if a pull request triggers a pipeline run and then the user pushes new commits into the pull request source branch, each push triggers a new copy of the pipeline run. If several pushes happen, several copies can run, which can consume excessive cluster resources.
```

// This module is included in the following assemblies:
// * pac/managing-pipeline-runs-pac.adoc

:_mod-docs-content-type: REFERENCE
[id="speccancel-pipeline-run-using-pipelines-as-code_{context}"]
= Annotations for specifying automatic cancellation-in-progress for a pipeline run

> By default, {pac} does not cancel pipeline runs automatically. Every pipeline run that {pac} creates and starts executes until it completes.
> 
> However, events that trigger pipeline runs can come in quick succession. For example, if a pull request triggers a pipeline run and then the user updates the pull request with a new commit, a new copy of the pipeline run starts every time. If several pushes happen, several copies can run, resulting in a drain on cluster resources.

Reviewer 2: ```suggestion
By default, {pac} does not cancel pipeline runs automatically. Every pipeline run that {pac} creates and starts executes until it completes. However, events that trigger pipeline runs can come in quick succession. For example, if a pull request triggers a pipeline run and then the user updates the pull request with a new commit, a new copy of the pipeline run starts as a result of each action. If several pushes happen, several copies can run, resulting in a drain on cluster resources.
```
removed the paragraph break and changed some words for clarity

You can configure a pipeline run to enable automatic cancellation-in-progress. If you enable automatic cancellation for a pipeline run, {pac} cancels the pipeline run in the following situations:

* {pac} has successfully started a copy of the same pipeline run for the same pull request or the same source branch.

* The pull request that triggered the pipeline run is merged or closed.

> You can use the following example to enable automatic cancellation for the `sample-pipeline` pipeline run:

Reviewer 2: ```suggestion
You can enable automatic cancellation for the `sample-pipeline` pipeline run by configuring the `metadata` in your [yaml file] as follows:
```
I'm not sure what file this is, but that's the point of the edit. :) It might be good to specify where the user needs to make this change.

Author: I tried to clarify a bit.
